lib/fs.js

```diff
@@ -35,7 +35,44 @@ const isWindows = process.platform === 'win32';
 
 const DEBUG = process.env.NODE_DEBUG && /fs/.test(process.env.NODE_DEBUG);
 const errnoException = util._errnoException;
-const printDeprecation = require('internal/util').printDeprecationMessage;
+
+var printDeprecation;
+try {
+  printDeprecation = require('internal/util').printDeprecationMessage;
+} catch (e) {
+  if (e.code !== 'MODULE_NOT_FOUND') throw e;
+
+  // TODO: remove this in master after 6.x
+  // This code was copied from internal/util and is required to give users
+  // a grace period before actually breaking modules that re-evaluate fs
+  // sources from context where internal modules are not allowed, e.g.
+  // older versions of graceful-fs module.
+
+  const prefix = `(${process.release.name}:${process.pid}) `;
+
+  printDeprecation = function(msg, warned) {
+    msg = `${prefix}${msg}`;
+    if (process.noDeprecation)
+      return true;
+
+    if (warned)
+      return warned;
+
+    if (process.throwDeprecation)
+      throw new Error(msg);
+    else if (process.traceDeprecation)
+      console.trace(msg.startsWith(prefix) ? msg.replace(prefix, '') : msg);
```

Comment:
@bnoordhuis I made this more clean here, but I don't think that the changes for `internal/util` need to go into this PR. Perhaps a separate one would be better.